utils.py

```diff
@@ -96,10 +104,14 @@ def get_nearest_neighbor(
         neighbor_norms = neighbors.norm(dim=-1)
     batch_dims = len(x.shape) - 1
     norms = neighbor_norms.repeat(*(1,)*batch_dims, 1) * x.norm(dim=-1).unsqueeze(-1)
+    zero_mask = norms == 0.0
     dots = (
         neighbors.unsqueeze(0).repeat(*(1,)*batch_dims, 1, 1) @ x.unsqueeze(-1)
     ).squeeze(-1)
+    distances = (dots / norms)
+    # Replace nan's
+    distances[zero_mask] = 0.0
```

Comment:
When do you get nan?
Should we add a warning message?

Reply:
Every time it's called because `<bos>` and `<pad>` have norm 0.

Reply:
I see! Makes a lot of sense.